### DISCUSSION TITLE
How do I build Storybook inside starter ui package?

### DISCUSSION THREAD

**Discussion by fibonacid** [vercel/turborepo#374] [Help] [answered]
### What version of Turborepo are you using?

1.0.17

### What version of Node are you using?

v14.17.6

### What version of Yarn are you using?

1.22.4

### Describe the Bug

Can't run `build-storybook` on ui package using turbo.

### Expected Behavior

Running build-storybook from ui package works fine.

```bash
yarn --cwd packages/ui build-storybook
```

Running build-storybook using turbo gets stuck forever.

```bash
yarn build-storybook
```

### To Reproduce

Clone [example repo](https://github.com/fibonacid/turborepo-build-storybook-bug)
```bash
git clone https://github.com/fibonacid/turborepo-build-storybook-bug
cd turborepo-build-storybook-bug
```

Install dependencies
```bash
yarn install
```

Run command
```bash
yarn build-storybook
```

**Comment by crutchcorn** [comment]
I'm experiencing the same in my project. Likewise with `start-storybook`. However, if I `cd` into the directory, it works as-intended

**Comment by fibonacid** [comment]
> I'm experiencing the same in my project. Likewise with `start-storybook`. However, if I `cd` into the directory, it works as-intended

If you put the start-storybook task in the turbo config with caching disabled works

**Comment by crutchcorn** [comment]
I have the following in my pipeline:

```
      "storybook": {
        "cache": false
      },
```

And it still seems to hang with the script:

```
    "storybook": "start-storybook -p 6007"
```

Is that what you meant @fibonacid ?

**Comment by rishabh2712** [comment]
Facing same issue. The thread is getting stuck at the start-storybook script, however running fine incase we gone down into the target folder.

**Comment by fibonacid** [comment]
> I have the following in my pipeline:
> 
> ```
>       "storybook": {
>         "cache": false
>       },
> ```
> 
> And it still seems to hang with the script:
> 
> ```
>     "storybook": "start-storybook -p 6007"
> ```
> 
> Is that what you meant @fibonacid ?

Yeah that works for me.

btw, the storybook versions i'm using are
```json
    "@storybook/addon-actions": "^6.4.9",
    "@storybook/addon-essentials": "^6.4.9",
    "@storybook/addon-links": "^6.4.9",
    "@storybook/react": "^6.4.9",
```

**Reply by crutchcorn** [reply]
These are mine, seems pretty similar:

```
"@storybook/addon-essentials": "^6.4.9",
"@storybook/builder-webpack5": "^6.4.9",
"@storybook/manager-webpack5": "^6.4.9",
"@storybook/react": "^6.4.9"
```

**Reply by 7iomka** [reply]
Share me please example repo with fixed problem if it fixed)
Thanks)

**Reply by fibonacid** [reply]
@7iomka Haven't found a fix for the `build-storybook` script yet. If anyone has a solution can submit a PR to the bug report repo

```bash
git clone https://github.com/fibonacid/turborepo-build-storybook-bug
```

**Reply by guillaumewuip** [reply]
@jaredpalmer as we have a reproductible repo, can this be converted back to an [issue](https://github.com/vercel/turborepo/issues/343)?

**Comment by fibonacid** [comment]
@crutchcorn Can you try to see if you can run start-storybook with turbo using the [example repo](https://github.com/fibonacid/turborepo-build-storybook-bug) i created?
Maybe your problem has something to do with your yarn or node version.

**Reply by crutchcorn** [reply]
I can reproduce this in this example repository. (Both for `start-storybook` and `build-storybook`)

For reference, I'm using `node@14.18.2` and yarn `1.22.17`.

I've also experienced this issue using `node@16.13.1`

What versions are you using?

**Comment by guillaumewuip** [comment]
I'm facing the same issue with a big storybook in one of my company's monorepo. 

I also have the same problem with another package that is built with webpack (I had to remove the `--progress` option to make it work) and another one that is stuck when I try to run tests (~4000 tests) with jest (don't found a fix yet). Turborepo is stuck forever at ~100% CPU.

Could it be related at how turborepo handle stdout / stderr? My three commands outputs a lot of text to the screen.

**Comment by kylemh** [comment]
Hey @jaredpalmer! Why was this moved to discussions? Storybook is going to be a pretty popular tool to integrate with and currently the out-of-box build script causes Turborepo to freeze.

**Comment by MangoChris** [comment]
I would even go one step further and say that for projects large enough to
need monorepo’s it’s practically standard.

On Thu, 23 Dec 2021 at 18:00, Kyle Holmberg ***@***.***>
wrote:

> Hey @jaredpalmer <https://github.com/jaredpalmer>! Why was this moved to
> discussions? Storybook is going to be a pretty popular tool to integrate
> with and currently the out-of-box build script causes Turborepo to freeze.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/turborepo/discussions/374#discussioncomment-1864702>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIQIGXCLLUNDGGR2EY3O4DUSNPSZANCNFSM5KO24GLA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
_________________

*Chris Brown*

*CEO*

***@***.*** ***@***.***>*

*Schedule a Call <https://calendly.com/mangomap-chris/intro/>*

*LinkedIn <https://uk.linkedin.com/in/chrisincambo>*

**Comment by 7iomka** [comment]
Move this to issues, please

**Comment by jaredpalmer** [comment]
This is likely due to either #336 or #292.

**Comment by jaredpalmer** [comment, marked as answer]
This is now being tracked in #422

**Comment by supaspoida** [comment]
Ahead of a fix, you could try the following flags w/your storybook commands:

```json
  "scripts": {
    "storybook": "start-storybook --ci -p 6006",
    "build-storybook": "build-storybook --quiet -o dist"
  }
```

The important bits are `--ci` and `--quiet`. I tried these changes on the reproduction repo and they worked for me.

**Reply by fibonacid** [reply]
@supaspoida Works for me!

**Reply by kylemh** [reply]
Although it's not needed for me in dev. Just build.

**Comment by leerob** [comment]
**Update:** We now have a design system example using:

- 🏎 [Turborepo](https://turborepo.org/) — High-performance build system for Monorepos
- 🚀 [React](https://reactjs.org/) — JavaScript library for user interfaces
- 🛠 [Tsup](https://github.com/egoist/tsup) — TypeScript bundler powered by esbuild
- 📖 [Storybook](https://storybook.js.org/) — UI component environment powered by Vite

https://github.com/vercel/turborepo/tree/main/examples/design-system

**Reply by kylemh** [reply]
Hey @leerob 

I wanted to suggest 2 design system examples. The one you have is great, but if I'm trying to avoid versioning/bundling by using `next-transpile-modules` in one of the consumers there are some things that should be done in the example to help any people attempting it. I outline the issues [here](https://github.com/vercel/turborepo/discussions/983) under "Storybook and consequences".

If 2 design system examples seems overkill, an alternative could simply be mentioned and discussed in the README of the example... maybe as a FAQ?

---

As an aside, the new design-system example has the documentation as an application that lives outside of the actual package. It feels like this would be slower than having the Storybook instance live _within_ the package itself since a bundle step would be required as-is.

**Comment by jnarowski** [comment]
Storybook 6.5 does not seem to be hitting the cache. I followed config from the design system exactly but it's missing the cache every time I build.

**Reply by kylemh** [reply]
should make a new issue no?
